File: SzachyPython/Figura.py
from Pozycja import Pozycja

class Figura(object):


    def __init__(self, pole, kolor):

        self.kolor = kolor
        self.pozycja = Pozycja(pole)

    def wypiszPole(self):
        self.pozycja.wypiszPole(self)

    def ruch(self, pole):
        x = ord(pole[0]) - ord('A') + 1
        y = ord(pole[1]) - ord('0')


        if(self.czyDozwolonyRuch(x,y) == False):
            print("Niepoprawne pole")
        else :
            self.pozycja.ustawXY(pole)

    def czyPoprawnePole(self, x, y):
        if (x < 1 or x > 8):
            return False
        if (y < 1 or y > 8):
            return False
        return True

    def czyDozwolonyDlaFigury(self, x, y):

        return False



    def czyDozwolonyRuch(self,x,y):
        if(self.czyPoprawnePole(x,y) == False):
            return False

        if(self.czyDozwolonyDlaFigury(x,y) == False):
            return False

        return True

File: szachyjava/src/Figura.java

public abstract class Figura {

    public Pozycja pozycja;
    public String nazwa;
    public String kolor;

    Figura(String pole){
        pozycja = new Pozycja(pole);
        // just Figura things....
    }

    public abstract void ruch(String pozycja);

    // metoda sprawdzająca czy można przesunąć figurę na wybrane pole
    // zawiera metody:
    // - czyPoprawnePole
    // - czyDozwolonyDlaFigury
    public abstract boolean czyDozwolonyRuch(int noweX, int noweY);

    // metoda sprawdzająca czy pole znajduje się na planszy
    public static boolean czyPoprawnePole(int noweX, int noweY){
        //czy mieści się na planszy
        if(noweX<1)
            return false;
        if(noweX>8)
            return false;
        if(noweY<1)
            return false;
        if(noweY>8)
            return false;

        return true;
    }

    // metoda sprawdzająca czy ruch dozwolony dla figury
    public abstract boolean czyDozwolonyDlaFigury(int noweX, int noweY, int x, int y);


    public boolean Equals(Figura innaFigura){

        if(pozycja.dajX() == innaFigura.pozycja.dajX()
                && pozycja.dajY() == innaFigura.pozycja.dajY()
                )
            return true;
        else
            return false;

    }

    // przeciazana wersja
    public boolean Equals(String innePole){

        if(pozycja.dajPozycje() == innePole)
            return true;
        else
            return false;

    }
}
